osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs

```diff
@@ -1,423 +1,82 @@
-            });
-        }
+            AddStep("clear lighten during break", () => lightenDuringBreaks.Value = false);
+            AddWaitStep("wait for dim", 5);
+            AddAssert($"is current dim {userDim}", () => container.DimEqual(userDim));
 
-        protected override void Dispose(bool isDisposing)
-        {
-            base.Dispose(isDisposing);
-            rulesets?.Dispose();
-        }
-
-        private class DummySongSelect : PlaySongSelect
-        {
-            protected override BackgroundScreen CreateBackground()
-            {
-                FadeAccessibleBackground background = new FadeAccessibleBackground(Beatmap.Value);
-                DimEnabled.BindTo(background.EnableUserDim);
-                return background;
-            }
-
-            public readonly Bindable<bool> DimEnabled = new Bindable<bool>();
-            public readonly Bindable<double> DimLevel = new Bindable<double>();
-            public readonly Bindable<double> BlurLevel = new Bindable<double>();
-
-            public new BeatmapCarousel Carousel => base.Carousel;
-
-            [BackgroundDependencyLoader]
-            private void load(OsuConfigManager config)
-            {
-                config.BindWith(OsuSetting.DimLevel, DimLevel);
-                config.BindWith(OsuSetting.BlurLevel, BlurLevel);
-            }
-
-            public bool IsBackgroundDimmed() => ((FadeAccessibleBackground)Background).CurrentColour == OsuColour.Gray(1 - (float)DimLevel.Value);
-
-            public bool IsBackgroundUndimmed() => ((FadeAccessibleBackground)Background).CurrentColour == Color4.White;
-
-            public bool IsUserBlurApplied() => ((FadeAccessibleBackground)Background).CurrentBlur == new Vector2((float)BlurLevel.Value * BackgroundScreenBeatmap.USER_BLUR_FACTOR);
-
-            public bool IsUserBlurDisabled() => ((FadeAccessibleBackground)Background).CurrentBlur == new Vector2(0);
-
-            public bool IsBackgroundInvisible() => ((FadeAccessibleBackground)Background).CurrentAlpha == 0;
-
-            public bool IsBackgroundVisible() => ((FadeAccessibleBackground)Background).CurrentAlpha == 1;
-
-            public bool IsBlurCorrect() => ((FadeAccessibleBackground)Background).CurrentBlur == new Vector2(BACKGROUND_BLUR);
-
-            /// <summary>
-            /// Make sure every time a screen gets pushed, the background doesn't get replaced
-            /// </summary>
-            /// <returns>Whether or not the original background (The one created in DummySongSelect) is still the current background</returns>
-            public bool IsBackgroundCurrent() => ((FadeAccessibleBackground)Background).IsCurrentScreen();
+            AddStep("clear break", () => isBreakTime.Value = false);
+            AddStep("set lighten during break", () => lightenDuringBreaks.Value = true);
+            AddWaitStep("wait for dim", 5);
+            AddAssert($"is current dim {userDim}", () => container.DimEqual(userDim));
```

Comment:
your tests don't really cover all scenarios. i'd split it out into four tests or so:

- turn on lighten mode, toggle break on, check, toggle break off, check
- toggle break on, check, turn on lighten mode, check
- turn on lighten mode, toggle break on, check, turn off lighten mode, check, toggle break off, check

or so